lmi_utils/image_utils/tiler.py

```diff
@@ -0,0 +1,216 @@
+    resize_h, resize_w = size
+
+    if mode == ScaleMode.PADDING:
+        pad_h = resize_h - image_h
+        pad_w = resize_w - image_w
+        image = F.pad(image, [0, pad_w, 0, pad_h])
+    elif mode == ScaleMode.INTERPOLATION:
+        image = F.interpolate(input=image, size=(resize_h, resize_w))
+    else:
+        msg = f"Unknown mode {mode}. Only padding and interpolation is available."
+        raise ValueError(msg)
+
+    return image
+
+
+@torch.inference_mode()
+def downscale_image(image: torch.Tensor, size: tuple, mode: ScaleMode = ScaleMode.PADDING) -> torch.Tensor:
+    """Opposite of upscaling. This image downscales image to a desired size.
+
+    Args:
+        image (torch.Tensor): Input image
+        size (tuple): Size to which image is down scaled.
+        mode (str, optional): Downscaling mode. Defaults to "padding".
+
+    Returns:
+        Tensor: Downscaled image
+    """
+    input_h, input_w = size
+    if mode == ScaleMode.PADDING:
+        image = image[:, :, :input_h, :input_w]
+    elif mode == ScaleMode.INTERPOLATION:
+        image = F.interpolate(input=image, size=(input_h, input_w))
+    else:
+        msg = f"Unknown mode {mode}. Only padding and interpolation is available."
+        raise ValueError(msg)
+
+    return image
+
+
+class Tiler:
+    logger = logging.getLogger('Tiler')
+    
+    def __init__(self, tile_size, stride):
+        """init tiler
+
+        Args:
+            tile_size (int | list): a int if tile_h equals to tile_w or a list of [tile_h, tile_w]
+            stride (int | list): a int if stride_h equals to stride_w or a list of [stride_h, stride_w]
+        """
+        if isinstance(tile_size, int):
+            tile_size = [tile_size]*2
+        if isinstance(stride, int):
+            stride = [stride]*2
+            
+        if not isinstance(tile_size, list) or len(tile_size)!=2:
+            raise Exception(f'tile size must be a list of two elements. Got: {tile_size}')
+        if not isinstance(stride, list) or len(stride)!=2:
+            raise Exception(f'stride must be a list of two elements. Got: {stride}')
+        if stride[0]>tile_size[0] or stride[1]>tile_size[1]:
+            raise Exception('Stride size must be smaller or equal to tile size')
+        
+        self.tile_size = tile_size
+        self.stride = stride
+        self.im_size: list
+        self.scale_size: list
+        self.batch_size: int
+        self.num_channel: int
+        self.n_tiles: list
+        
+        
+    @classmethod
+    def from_pt(cls, pt_path):
+        """init tiler from a pt file
+
+        Args:
+            pt_path (str): path to the metadata pt file
+        """
+        obj = cls(0,0) # init an obj using dummy sizes
+        metadata = torch.load(pt_path)
+        for k,v in metadata.items():
+            setattr(obj,k,v)
+        return obj
+        
+    
+    @torch.inference_mode()
+    def tile(self, im:torch.Tensor, mode=ScaleMode.PADDING) -> torch.Tensor:
```

Comment:
should also add support for overlap ratio.

Reply:
It supports overlap ratio. You can specify the stride paramater in the Tiler init function.